### PR TITLE
EzCheckbox Redesign

### PR DESCRIPTION
**Pull request by noranda**
_**Note**: This is currently set to merge into `nb/ezradio`, but should eventually be merged into `v16` once `nb/ezradio` is merged into `main` and `v16` is rebased._

_**Note**: Thanks to all the initial setup work from @maggiedbaker! This work will close https://github.com/ezcater/recipe/pull/859._

## What did we change?
- Implements `EzCheckbox` redesign including MUI implementation, documentation, and regression tests.
- Moves legacy `EzCheckbox` to separate implementation with `legacy` tag.
- Adds `EzFormGroup` component to Recipe to handle groups of checkboxes.
- Deprecates `EzField` with `type="checkbox"` and links to `EzCheckbox`.
- Adds new migration guide for v16.

## Why are we doing this?
This work implements the Checkbox redesign for both fulfillment and marketplace domains.

Designs found [here](https://www.figma.com/file/mRXuf1mxiFCMOMWA2oKTdN/ezCater-DS?node-id=562%3A454).

## Screenshot(s) / Gif(s):
Screen Shot 2022-09-13 at 11 03 19 AM
Screen Shot 2022-09-13 at 11 03 32 AM
Screen Shot 2022-09-13 at 11 03 44 AM
Screen Shot 2022-09-13 at 11 08 29 AM
Screen Shot 2022-09-13 at 11 08 37 AM

## Checklist

- [x] Provide test coverage for the changes made
- [x] Create a changeset (`yarn changeset`) with a summary of the changes

Thanks to all the setup work from @maggiedbaker! This work will close https://github.com/ezcater/recipe/pull/859.

### CODE REVIEW

File: packages/recipe/src/components/EzCheckbox/EzCheckbox.md [hunk 1]
```diff
@@ -3,101 +3,403 @@ name: EzCheckbox
...
+      </EzLayout>
+
+      <EzLayout style={{backgroundColor: '#034a34', color: 'white', padding: '0 16px'}}>
+        <EzFormControlLabel
+          control={<EzCheckbox defaultChecked variant="filled" />}
+          label="Filled"
+        />
+      </EzLayout>
+
+      <EzLayout style={{backgroundColor: '#034a34', color: 'white', padding: '0 16px'}}>
```
Comment (danidewitt) [inline packages/recipe/src/components/EzCheckbox/EzCheckbox.md:78]: same comment as the other PR about including inline styles in our documenation

File: packages/recipe/src/components/EzCheckbox/EzCheckbox.md [hunk 2]
```diff
@@ -3,101 +3,403 @@ name: EzCheckbox
...
+        checked={acknowledgementChecked}
+      />
+    </EzLayout>
   );
 };
 ```
 
 ---
 
+## Custom Styles
```
Comment (danidewitt) [inline packages/recipe/src/components/EzCheckbox/EzCheckbox.md:295]: same question as the other PR about including overrides

File: packages/recipe/src/components/EzCheckbox/Implementations/EzCheckboxLegacy.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+              d="M6.525 11.576L3.142 8.214A.494.494 0 0 1 3 7.849c0-.149.047-.27.142-.365l.75-.729a.452.452 0 0 1 .354-.162c.142 0 .267.054.375.162l2.268 2.269 4.862-4.862A.515.515 0 0 1 12.126 4c.142 0 .26.054.354.162l.75.73a.494.494 0 0 1 .142.364c0 .149-.048.27-.142.365l-5.976 5.955a.462.462 0 0 1-.365.162.462.462 0 0 1-.364-.162z"
+            />
+          </g>
+        </svg>
+      </span>
+    );
+  }
+);
+
+const EzCheckboxLegacy: FC<EzCheckboxLegacyProps & RefAttributes<any>> = forwardRef(
```
Comment (danidewitt) [inline packages/recipe/src/components/EzCheckbox/Implementations/EzCheckboxLegacy.tsx:92]: nice job pulling the legacy stuff out into its own implementation! 🎉

**Review by danidewitt — APPROVED** [review]
basically same questions as the radio PR. nice job with the legacy implementation!

File: packages/recipe/src/components/EzCheckbox/EzCheckbox.md (hunk 2 above)
Comment (noranda) [reply packages/recipe/src/components/EzCheckbox/EzCheckbox.md:295]: @danidewitt Same [answer](https://github.com/ezcater/recipe/pull/861#discussion_r972199064)!

File: packages/recipe/src/components/EzCheckbox/Implementations/EzCheckboxLegacy.tsx (hunk 3 above)
Comment (noranda) [reply packages/recipe/src/components/EzCheckbox/Implementations/EzCheckboxLegacy.tsx:92]: Thanks to @maggiedbaker for doing this!

File: packages/recipe/src/components/EzCheckbox/EzCheckbox.md (hunk 1 above)
Comment (noranda) [reply packages/recipe/src/components/EzCheckbox/EzCheckbox.md]: @danidewitt Changed all EzLayouts with styles to divs!

**Review by danidewitt — APPROVED** [review]

**Review by pgrippi — DISMISSED** [review]

**Review by pgrippi — APPROVED** [review]

**Review by danidewitt — APPROVED** [review]

**Pull request merged**